util/uvmdvgen/env_cfg.sv.tpl

```diff
@@ -21,9 +21,11 @@ class ${name}_env_cfg extends dv_base_env_cfg #(.RAL_T(${name}_reg_block));
 
   `uvm_object_new
 
+% if has_ral:
   virtual function void initialize_csr_addr_map_size();
     this.csr_addr_map_size = ${name.upper()}_ADDR_MAP_SIZE;
   endfunction : initialize_csr_addr_map_size
+% endif
```

Comment:
See  more references to ral on line 40 of this file below.

Reply:
Should we make the assumption that if a DUT uses the CIP library, then the DUT will have CSRs? The section on line 40 that you refer to already is gated by a 'if is_cip' conditional, so the addition of the check 'if has_ral' would give the assumption that a block can be CIP without having CSRs.

Reply:
Yes, agree.